### PR TITLE
feat(STONEINTG-1072): removing sidecard and implement it by Clamav DB

### PR DESCRIPTION
**Pull request by kasemAlem**
# Before you complete this pull request ...

Look for any open pull requests in the repository with the title "e2e-tests update" and 
see if there are recent e2e-tests updates that will be applicable to your change.

### CODE REVIEW

File: task/clamav-scan/0.2/clamav-scan.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+      description: The name of the ConfigMap to read CA bundle data from.
+      default: trusted-ca
+    - name: ca-trust-config-map-key
+      type: string
+      description: The name of the key in the ConfigMap that contains the CA bundle data.
+      default: ca-bundle.crt
+
+  steps:
+    - name: extract-and-scan-image
+      image: quay.io/kasem_org/clamav-db-test:v14
```
Comment (jsztuka) [inline task/clamav-scan/0.2/clamav-scan.yaml:37]: This will be updated once https://github.com/konflux-ci/konflux-test/pull/328 see the light of the day.

Comment (kasemAlem) [reply task/clamav-scan/0.2/clamav-scan.yaml:37]: @jsztuka 
[konflux-ci/konflux-test/pull/328](https://github.com/konflux-ci/konflux-test/pull/328)  was just merged, 
should I update the image value now or we need to wait for tomorrow's first image ?

**Comment by kasemAlem** [comment]
check this newer run log,  duration took 4 minutes

```
step-extract-and-scan-image

Extracting image(s).
Running "oc image extract" on image of arch amd64
Scanning image for arch amd64. This operation may take a while.
LibClamAV Warning: Max file-size was set to 4293918720 bytes. Unfortunately, scanning files greater than 2147483647 bytes (2 GiB - 1) is not supported.

----------- SCAN SUMMARY -----------
Known viruses: 8700659
Engine version: 1.0.7
Scanned directories: 2933
Scanned files: 24155
Infected files: 0
Data scanned: 1051.98 MB
Data read: 880.60 MB (ratio 1.19:1)
Time: 188.728 sec (3 m 8 s)
Start Date: 2024:12:02 18:25:08
End Date:   2024:12:02 18:28:17
Executed-on: Scan was executed on clamscan version - ClamAV 1.0.7/27474/Sun Dec  1 09:42:26 2024 Database version: 27474
[
	{
		"filename": "/work/logs/clamscan-result-log-amd64.json",
		"namespace": "required_checks",
		"successes": 2
	}
]
{"timestamp":"1733164097","namespace":"required_checks","successes":2,"failures":0,"warnings":0,"result":"SUCCESS","note":"All checks passed successfully"}
{"timestamp":"1733164097","namespace":"required_checks","successes":2,"failures":0,"warnings":0,"result":"SUCCESS","note":"All checks passed successfully"}
{"timestamp":"1733164097","namespace":"required_checks","successes":2,"failures":0,"warnings":0,"result":"SUCCESS","note":"All checks passed successfully"}
{"image": {"pullspec": "quay.io/redhat-user-workloads/kalem-tenant/devfile-sample-python-basic:3fe0848df23095703ded201abf40026a6648330f", "digests": ["sha256:70a9c298dccedbde0dfc3b679807db118cf2152ee63b779f95ca4267828bb0b8"]}}
step-upload

Selecting auth
Using token for quay.io/redhat-user-workloads/kalem-tenant/devfile-sample-python-basic
Attaching to quay.io/redhat-user-workloads/kalem-tenant/devfile-sample-python-basic:3fe0848df23095703ded201abf40026a6648330f
Uploading 4f523535a2d9 clamscan-ec-test-amd64.json
Uploading 0476087e82b1 clamscan-result-amd64.log
Uploaded  4f523535a2d9 clamscan-ec-test-amd64.json
Uploaded  0476087e82b1 clamscan-result-amd64.log
Attached to [registry] quay.io/redhat-user-workloads/kalem-tenant/devfile-sample-python-basic@sha256:70a9c298dccedbde0dfc3b679807db118cf2152ee63b779f95ca4267828bb0b8
Digest: sha256:37ed439a4d59e4959913186258e720344f3badd831607c27863e31369f9d6613
```

**Review by dirgim — APPROVED** [review]
LGTM

File: task/clamav-scan/0.2/MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Migration from 0.1 to 0.2
+
+Version 0.2:
+
+On this version the sidecard is removed from the task and required tools (jq, oc ..) were added to the Clamav BD container image
```
Comment (jsztuka) [inline task/clamav-scan/0.2/MIGRATION.md:5]: sidecard > sidecar

File: task/clamav-scan/0.2/clamav-scan.yaml [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      description: The name of the ConfigMap to read CA bundle data from.
+      default: trusted-ca
+    - name: ca-trust-config-map-key
+      type: string
+      description: The name of the key in the ConfigMap that contains the CA bundle data.
+      default: ca-bundle.crt
+
+  steps:
+    - name: extract-and-scan-image
+      image: quay.io/konflux-ci/clamav-db:latest
```
Comment (jsztuka) [inline task/clamav-scan/0.2/clamav-scan.yaml:37]: Can we just add comment below about that this floating tag is being build daily, hence the interpretation.

**Review by jsztuka — APPROVED** [review]
lgtm

**Pull request merged**